pkg/hub/user_proxy.go

```diff
@@ -0,0 +1,197 @@
+package hub
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/open-cluster-management/cluster-proxy-addon/pkg/config"
+	"github.com/open-cluster-management/cluster-proxy-addon/pkg/helpers"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	addonv1alpha1client "open-cluster-management.io/api/client/addon/clientset/versioned"
+)
+
+type UserServerOptions struct {
+	proxyUdsName string
+	serverCert   string
+	serverKey    string
+	serverPort   int
+	addonClient  addonv1alpha1client.Interface
+}
+
+func NewUserServerOptions() *UserServerOptions {
+	return &UserServerOptions{}
+}
+
+func (u *UserServerOptions) AddFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+	flags.StringVar(&u.proxyUdsName, "proxy-uds", u.proxyUdsName, "the UDS name to connect to")
+	flags.StringVar(&u.serverCert, "server-cert", u.serverCert, "Secure communication with this cert")
+	flags.StringVar(&u.serverKey, "server-key", u.serverKey, "Secure communication with this key")
+	flags.IntVar(&u.serverPort, "server-port", u.serverPort, "handle user request using this port")
+}
+
+func (u *UserServerOptions) Validate() error {
+	if u.proxyUdsName == "" {
+		return fmt.Errorf("The proxy-uds is required")
+	}
+
+	if u.serverCert == "" {
+		return fmt.Errorf("The server-cert is required")
+	}
+
+	if u.serverKey == "" {
+		return fmt.Errorf("The server-key is required")
+	}
+
+	if u.serverPort == 0 {
+		return fmt.Errorf("The server-port is required")
+	}
+
+	return nil
+}
+
+func (u *UserServerOptions) Handler(wr http.ResponseWriter, req *http.Request) {
+	if u.addonClient == nil {
+		http.Error(wr, fmt.Sprintf("addon client is nil"), http.StatusInternalServerError)
+		return
+	}
+
+	if klog.V(4).Enabled() {
+		dump, err := httputil.DumpRequest(req, true)
+		if err != nil {
+			http.Error(wr, err.Error(), http.StatusBadRequest)
+			return
+		}
+		klog.V(4).Infof("request:\n%s", string(dump))
+	}
+
+	// parse clusterID from current requestURL
+	clusterID, kubeAPIPath, err := helpers.ParseRequestURL(req.RequestURI)
+	if err != nil {
+		http.Error(wr, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// check if cluster-proxy addon installed in target managed cluster
+	clusterProxyAddon, err := u.addonClient.AddonV1alpha1().ManagedClusterAddOns(clusterID).Get(context.TODO(), config.ADDON, v1.GetOptions{})
+	if err != nil {
+		http.Error(wr, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var manifestApplied = false
+	for _, c := range clusterProxyAddon.Status.Conditions {
```

Comment:
you can use `meta.IsStatusConditionTrue` instead of this

Reply:
OK, got it

Reply:
Using IsStatusConditionFalse to check error